spec/discovery-indexer/reader/purlxml_parser_strict_spec.rb

```diff
@@ -177,19 +179,37 @@
       expect(pm.send(:parse_file_ids)).to be_nil
+      expect(pm.send(:parse_sw_image_ids)).to be_nil
+      expect(pm.send(:parse_sw_file_ids)).to be_nil
     end
 
-    it 'returns nil when content metadata is present but no image or page ids are present' do
+    it 'returns nil when content metadata is present but no image, page, or thumb resource types are present' do
+      public_xml_no_content = "<publicObject id='druid:aa111aa1111'>#{@rights_metadata}#{@identity_metadata}#{@blank_content_metadata}</publicObject>"
+      pm = described_class.new('', Nokogiri::XML(public_xml_no_content))
+      expect(pm.send(:parse_image_ids)).to be_nil
+      expect(pm.send(:parse_sw_image_ids)).to be_nil
+    end
+
+    it 'returns nil when content metadata is present but no ids are present in the file tags' do
       public_xml_no_content = "<publicObject id='druid:aa111aa1111'>#{@rights_metadata}#{@identity_metadata}#{@blank_content_metadata}</publicObject>"
       pm = described_class.new('', Nokogiri::XML(public_xml_no_content))
-      expect(pm.send(:parse_image_ids)).to be_empty
       expect(pm.send(:parse_file_ids)).to be_nil
+      expect(pm.send(:parse_sw_file_ids)).to be_nil
+    end
+
+    it 'returns all druids and ids from the resource/file and objectIds and fileIds from resource/externalFile tag regardless of resource type when present' do
+      pm = described_class.new('tn629pk3948', @available_purl_xml_ng_doc)
+      expect(pm.send(:parse_sw_file_ids)).to eq(["tn629pk3948%2Ftn629pk3948_sl.mp4", "tn629pk3948%2Ftn629pk3948_img_1.jp2", "tn629pk3948%2Ftn629pk3948_img_2.jp2", "tn629pk3948%2Ftn629pk3948_img_3.jp2", "tn629pk3948%2Ftn629pk3948_pg_1.pdf", "tn629pk3948%2Ftn629pk3948_pg_1.jp2", "tn629pk3948%2Ftn629pk3948_pg_1.pdf", 'tn629pk3948%2Ftn629pk3948_thumb_1.jp2', 'tn629pk3948%2Ftn629pk3948_thumb_2.jp2'])
+    end
+
+    it 'returns ids from resource/file tag when present if resource type is image, page, or thumb' do
+      pm = described_class.new('tn629pk3948', @available_purl_xml_ng_doc)
+      expect(pm.send(:parse_image_ids)).to eq(["tn629pk3948_img_1.jp2", "tn629pk3948_img_2.jp2", "tn629pk3948_img_3.jp2", "tn629pk3948_pg_1.jp2", "tn629pk3948_thumb_1.jp2", "tn629pk3948_thumb_2.jp2"])
     end
 
-    it 'returns image and page ids when present' do
-      pm = described_class.new('', @available_purl_xml_ng_doc)
-      expect(pm.send(:parse_image_ids)).to eq(['tn629pk3948_img_1.jp2', 'tn629pk3948_img_2.jp2', 'tn629pk3948_img_3.jp2', 'tn629pk3948_pg_1.jp2', 'tn629pk3948_thumb_1.jp2', 'tn629pk3948_thumb_2.jp2'])
-      expect(pm.send(:parse_file_ids)).to eq(["tn629pk3948_sl.mp4", "tn629pk3948_img_1.jp2", "tn629pk3948_img_2.jp2", "tn629pk3948_img_3.jp2", "tn629pk3948_pg_1.pdf", "tn629pk3948_pg_1.jp2", "tn629pk3948_pg_1.pdf", 'tn629pk3948_thumb_1.jp2', 'tn629pk3948_thumb_2.jp2'])
+    it 'returns druids and ids from the resource/file tag and objectIds and fileIds from resource/externalFile tag if resource type is image, page, or thumb' do
```

Comment:
This test is really about the parse_sw_image_ids  method ... so it would be better, IMO, for the description to reflect that.  It's appropriate for tests to be for the object as a whole as well as methods in the object and we've got lots of places in DLSS code where we do that.